src/uwtools/config/tools.py

```diff
@@ -105,10 +103,10 @@ def realize_config(
         if isinstance(input_config, Config)
         else format_to_config(input_format)(config=input_config)
     )
-    input_obj.dereference()
-    input_obj = _realize_config_update(input_obj, values, values_format)
+    if supplemental_configs:
+        input_obj = _realize_config_update(input_obj, supplemental_configs)
+        input_obj.dereference()
```

Comment:
It might not hurt to do all the updates, then dereference once after all the updates are complete.

Reply:
I'm not sure I understand. `_realize_config_update()` is taking in the entire list of `supplemental_configs`, updating all of the values within `_realize_config_update()` and only dereferencing once, upon return to `realize_config()`. Would you prefer `dereference()` to be elsewhere in the function?

Reply:
Will dereferencing get done if there are no supplemental configs? Should we outdent the `input_obj.dereference()` line so that it executes no matter what?

Reply:
It did not, so I moved it outside of the if statement and have also changed the tests in a way that should catch that in the future.

Reply:
My haste in reviewing led me to read it as a loop instead of an if-statement. Sorry for the confusion.